presto-common/src/main/java/com/facebook/presto/common/zorder/ZOrder.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.common.zorder;
+
+import com.facebook.presto.common.type.Type;
+
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.List;
+
+public final class ZOrder
+{
+    private ZOrder()
+    {
+    }
+
+    public static int highestSetBit(int value)
+    {
+        int count = 0;
+        while (value != 0) {
+            value >>= 1;
+            count++;
+        }
+        return count;
```

Comment:
`
int position = 1;
while (value != 0) {
    value >> 1;
    position++;
}
return --position;
`

When you use result from this function, the return value means the position of the highest set bit (0-based indexing).